src/main/java/com/arpnetworking/tsdcore/model/CompositeUnit.java

```diff
@@ -0,0 +1,338 @@
+/**
+ * Copyright 2016 Groupon.com
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.arpnetworking.tsdcore.model;
+
+import com.google.common.base.Optional;
+import com.google.common.collect.ImmutableMap;
+import net.sf.oval.constraint.NotNull;
+
+
+/**
+ * Compound Unit used in 2g file format.
```

Comment:
I think that's exactly where the disconnect is. The broken format (unit + scale) is a serialization format. There is no need, at least at the moment, to undertake a large refactor of the internal Unit class. In fact, this is precisely why we separate the internal and external data models; so changes to one do not ripple to the other.

Instead I strongly recommend that we simply map the 2G Unit into the existing internal data model. It would be nice if you added the missing units to the internal model since the 2G format is more expressive. This would be a simple task, in fact a Python or Bash code generator should be able to whip up the change in no time (otherwise you'll need to do some cut-and-paste or get Gil or Matt's help with editor macros).

The other points is the naming inside the model class, please don't call something "Unit2G" since it's already within the 2G Version class. These are all encapsulated and further disambiguation via naming is unnecessary.